### PR TITLE
Added api link for /profile

### PR DESCRIPTION
**Pull request by jimmyl02**
Added getCleanedChallenge() in backend and linked frontend and backend.

### CODE REVIEW

File: client/src/pages/profile.js [hunk 1]
```diff
@@ -21,16 +22,53 @@ export default withStyles({
...
+      .then(data => {
+        this.setState({
+          name: data.name,
+          division: data.division,
+          score: data.score.score,
+          teamToken: data.teamToken,
+          solves: data.returnedSolves
+        })
+        let placementStr = String(data.score.place)
+        if (data.score.place >= 11 && data.score.place <= 13) {
```
Comment (chen-robert) [inline client/src/pages/profile.js]: This feels like it could go into a utility method

File: server/responses/index.js [hunk 2]
```diff
@@ -102,6 +102,14 @@ const responseList = {
   goodChallenges: {
     status: 200,
     message: 'The retrieval of challenges was successful'
+  },
+  goodUserData: {
+    status: 200,
+    message: 'The user was successfully retrieved'
+  },
+  badUserData: {
```
Comment (chen-robert) [inline server/responses/index.js]: Is it possible to provide a more in-depth reason for why it failed to retrieve? badUserData sounds like the user does not exist

File: server/api/users-me.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const user = await db.users.getUserByUserId({ userid: uuid })
+
+    if (user === undefined) return responses.badUserData
+
+    const solves = await db.solves.getSolvesByUserId({ userid: uuid })
+    const teamToken = await auth.token.getToken(auth.token.tokenKinds.team, uuid)
+    const score = await cache.leaderboard.getScore({ id: uuid })
+    const returnedScore = score === null ? 0 : score
+
+    const returnedSolves = []
```
Comment (chen-robert) [inline server/api/users-me.js]: This should just be called `solves`?

File: client/src/pages/profile.js (hunk 1 above)
Comment (jimmyl02) [reply client/src/pages/profile.js]: This sounds right, I will move it.

File: test/integration/profile.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  name: uuidv4(),
+  division: Object.values(config.divisions)[0]
+}
+
+test.serial('succeeds with goodUserData', async t => {
+  config.verifyEmail = false
+
+  let resp = await request(app)
+    .post(process.env.API_ENDPOINT + '/auth/register')
+    .send({
```
Comment (chen-robert) [inline test/integration/profile.js]: Small thing, can't we just do `.send(testUser)` here?

**Pull request merged**